### ISSUE TITLE
Keyboard Shortcut for "Show Desktop"

### ISSUE THREAD

**Issue by ThanosApostolou** [jderose9/dash-to-panel#105]
I really like your "Show Desktop" implementation which allows windows to be restored, in contrast with when you set a Keyboard Shortcut for Gnome's built in "Hide all normal windows".

So, it would be really nice if you had a configurable keyboard shortcut for that (maybe with default <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>d</kbd> or <kbd>Super</kbd>+<kbd>d</kbd>) easily accessible from the extension's settings.

**Comment by jderose9** [comment]
Great idea, thanks!

**Comment by DarthWound** [comment]
@ThanosApostolou Hello, for me `super`+`d` works the same than the "show desktop" button, it restores the windows. Fedora25+Gnome3.22+Wayland.

**Comment by ThanosApostolou** [comment]
@DarthWound Wow you are right. Indeed when I set <kbd>Super</kbd>+<kbd>d</kbd> it works, but when I set <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>d</kbd> it doesn't (tested on Manjaro with Xorg). Maybe it's a Gnome bug.

**Comment by DarthWound** [comment]
@ThanosApostolou I tried with other shortcuts like <kbd>ctrl</kbd>+<kbd>alt</kbd>+<kbd>d</kbd> after your last reply and it works too on my computer.

**Comment by jderose9** [comment]
Both keyboard combinations are working for me on Fedora 25/3.22/Xorg.

I didn't realize the restore functionality was built into gnome. I'm thinking that the behavior from dash-to-panel button should be hooked into gnome's function rather that my implementation, because right now if you use them together, they get out of sync (If you use the key combo to show desktop and then hit the button it will attempt to minimize all windows again rather than restore).

**Comment by DarthWound** [comment]
Yeah, if it doesn't work for @ThanosApostolou there's an issue somewhere else, I tried under Xorg too and shortcuts still work. Imtegrating this function in DtP seems to be a bad idea as you said @jderose9, however some people are unaware of this function in GNOME as it's not enabled by default, so at least it could be nice to mention it in the README as a tip/trick or something.

**Comment by ThanosApostolou** [comment]
It was probably my fault. Maybe it had something to do with the fact that I was using nemo to handle the desktop icons because I didn't like nautilus. Indeed though, it would be better to be in sync with gnome's built in functionality. Maybe you can still add the ability to change the keyboard shortcut from your extension's settings by making it change the Gnome's settings (I think keyboard shortcuts are handled in dconf). Otherwise, you can mark this as closed.

**Comment by AlexGluck** [comment]
In gnome that's already exist

![Снимок экрана от 2021-11-06 23-30-38](https://user-images.githubusercontent.com/8583390/140622993-2c9704ec-6d35-4450-a7f6-f34ad88c8d5f.png)

![Снимок экрана от 2021-11-06 23-30-58](https://user-images.githubusercontent.com/8583390/140622992-e1dcde90-424f-40f5-9da9-ebb8aaafa8d0.png)
